konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/step/ml/ssd/regression/RegressionOutputRunner.java

```diff
@@ -0,0 +1,160 @@
+ *  * terms of the Apache License, Version 2.0 which is available at
+ *  * https://www.apache.org/licenses/LICENSE-2.0.
+ *  *
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  * License for the specific language governing permissions and limitations
+ *  * under the License.
+ *  *
+ *  * SPDX-License-Identifier: Apache-2.0
+ *  *****************************************************************************
+ */
+
+package ai.konduit.serving.pipeline.impl.step.ml.ssd.regression;
+
+import ai.konduit.serving.annotation.runner.CanRun;
+import ai.konduit.serving.pipeline.api.context.Context;
+import ai.konduit.serving.pipeline.api.data.Data;
+import ai.konduit.serving.pipeline.api.data.NDArray;
+import ai.konduit.serving.pipeline.api.data.NDArrayType;
+import ai.konduit.serving.pipeline.api.data.ValueType;
+import ai.konduit.serving.pipeline.api.step.PipelineStep;
+import ai.konduit.serving.pipeline.api.step.PipelineStepRunner;
+import ai.konduit.serving.pipeline.util.DataUtils;
+import lombok.AllArgsConstructor;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.*;
+
+@AllArgsConstructor
+@CanRun(RegressionOutputStep.class)
+public class RegressionOutputRunner implements PipelineStepRunner {
+
+    protected final RegressionOutputStep step;
+
+
+    @Override
+    public void close() {
+
+    }
+
+    @Override
+    public PipelineStep getPipelineStep() {
+        return step;
+    }
+
+    @Override
+    public Data exec(Context ctx, Data data) {
+
+        String inputName = step.inputName();
+
+
+        if (inputName == null) {
+            String errMultipleKeys = "NDArray field name was not provided and could not be inferred: multiple NDArray fields exist: %s and %s";
+            String errNoKeys = "NDArray field name was not provided and could not be inferred: no image NDArray exist";
+            inputName = DataUtils.inferField(data, ValueType.NDARRAY, false, errMultipleKeys, errNoKeys);
+        }
+
+
+        NDArray regressionOutput = data.getNDArray(inputName);
+        if (regressionOutput.shape().length > 2) {
+            throw new UnsupportedOperationException("Invalid input to ClassifierOutputStep: only rank 1 or 2 inputs are available, got array with shape" + Arrays.toString(regressionOutput.shape()));
+
+        }
+        regressionOutput = FloatNDArrayToDouble(regressionOutput);
+
+
+        boolean batch = false;
+        if (regressionOutput.shape().length == 2 && regressionOutput.shape()[0] > 1) {
+            batch = true;
+        }
+
+        Map<String, Integer> outputNames = step.names();
+
+        if (outputNames == null || outputNames.isEmpty()) {
+            throw new UnsupportedOperationException(".names field was not provided or equal null");
```

Comment:
Let's be more explicit about the step: `"RegressionOutputStep names field was not provided or is null"`